tests/scripts/ci.py

```diff
@@ -434,6 +436,26 @@ def cli_name(s: str) -> str:
     return s.replace("_", "-")
 
 
+def typing_get_origin(annotation):
+    if sys.version_info >= (3, 8):
+        return typing.get_origin(annotation)
+    else:
+        return annotation.__origin__
```

Comment:
Thanks for the fix! Python's treatment of `Optional` is pretty annoying and has changed at some point between 3.7 and 3.10 (moved from an alias to `Union` to its own thing), I noticed on 3.7 this fails though:

```
Traceback (most recent call last):
  File "tests/scripts/ci.py", line 681, in <module>
    main()
  File "tests/scripts/ci.py", line 657, in main
    add_subparser(func, subparsers)
  File "tests/scripts/ci.py", line 506, in add_subparser
    is_optional = is_optional_type(value.annotation)
  File "tests/scripts/ci.py", line 456, in is_optional_type
    return (typing_get_origin(annotation) == typing.Union) and (
  File "tests/scripts/ci.py", line 445, in typing_get_origin
    return annotation.__origin__
AttributeError: type object 'bool' has no attribute '__origin__'
```

Reply:
Thank you!  It looks like python3.9+ represent it as `typing.Optional`, while earlier versions give `typing.Union`.  After some quick testing, checking for `__origin__` works on all versions.  Unfortunately, I don't see any public classes for checking if the annotation is any `typing.*` class, as the `typing._Final` class is private.